RockWeb/App_Code/GetEventCalendarFeed.cs

```diff
@@ -115,70 +117,87 @@ private iCalendar CreateICalendar( CalendarProps calendarProps )
-                        foreach ( var a in eventItem.EventItemAudiences )
+                        foreach ( var icalEvent in ical[0].Events )
                         {
-                            ievent.Categories.Add( a.DefinedValue.Value );
+                            // We get all of the schedule info from Schedule.iCalendarContent
+                            Event ievent = icalEvent.Copy<Event>();
+
+                            ievent.Summary = !string.IsNullOrEmpty( eventItem.Name ) ? eventItem.Name : string.Empty;
+                            ievent.Location = !string.IsNullOrEmpty( occurrence.Location ) ? occurrence.Location : string.Empty;
+
+                            ievent.DTStart.SetTimeZone( icalendar.TimeZones[0] );
+                            ievent.DTEnd.SetTimeZone( icalendar.TimeZones[0] );
+
+                            // Rock has more descriptions than iCal so lets concatenate them
+                            string description = CreateEventDescription( eventItem, occurrence );
+
+                            // Don't set the description prop for outlook to force it to use the X-ALT-DESC property which can have markup.
+                            if ( interactionDeviceType != "Outlook" )
+                            {
+                                ievent.Description = description.ConvertBrToCrLf()
+                                                                    .Replace( "</P>", "" )
+                                                                    .Replace( "</p>", "" )
+                                                                    .Replace( "<P>", Environment.NewLine )
+                                                                    .Replace( "<p>", Environment.NewLine )
+                                                                    .Replace( "&nbsp;", " " )
+                                                                    .SanitizeHtml();
+                            }
+
+                            // HTML version of the description for outlook
+                            ievent.AddProperty( "X-ALT-DESC;FMTTYPE=text/html", "<html>" + description + "</html>" );
+
+                            // classification: "PUBLIC", "PRIVATE", "CONFIDENTIAL"
+                            ievent.Class = "PUBLIC";
+
+                            if ( !string.IsNullOrEmpty( eventItem.DetailsUrl ) )
+                            {
+                                Uri result;
+                                if ( Uri.TryCreate( eventItem.DetailsUrl, UriKind.Absolute, out result ) )
+                                {
+                                    ievent.Url = result;
+                                }
+                                else if ( Uri.TryCreate( "http://" + eventItem.DetailsUrl, UriKind.Absolute, out result ) )
+                                {
+                                    ievent.Url = result;
+                                }
+                            }
```

Comment:
The above code checks for a valid URL, attempts to fix the URL if it isn't valid, and ignores an unfixable invalid URL. Previously a url like www.rockrms.com would crash the feed.

Reply:
@chrismcgrath Ok.  My only last concern is the number of schedules that would be updated in that loop.  I've checked a few large systems and have seen between 2287 and 16380  schedules (`SELECT COUNT(1) FROM [Schedule] WHERE [IsActive] = 1`).  Can that Queryable in the migration Up() be tightened/target any further (to only the necessary schedules)?